src/bidiMapper/domains/context/browsingContextProcessor.ts

```diff
@@ -187,6 +187,28 @@ export class BrowsingContextProcessor {
         this.#browsingContextStorage,
         this.#logger
       );
+
+      // Get all top-level context scripts.
+      const scripts = this.#preloadScriptStorage.findPreloadScripts({
+        contextId: null,
+      });
+
+      // Add them all to the new context.
+      for (const script of scripts) {
+        const functionDeclaration = script.functionDeclaration;
+        const sandbox = script.sandbox;
+
+        // The spec provides a function, and CDP expects an evaluation.
+        const cdpPreloadScriptId = await context.cdpTarget.addPreloadScript(
```

Comment:
There will be a race condition between [`CdpTarget.create` -> `void cdpTarget.#unblock()`](https://github.com/GoogleChromeLabs/chromium-bidi/blob/43882f46dca6e203b0bdbb12c4aa9d79a6a3895b/src/bidiMapper/domains/context/cdpTarget.ts#L55) and adding preload script. Adding preload script should be a part of [`async #unblock()`](https://github.com/GoogleChromeLabs/chromium-bidi/blob/43882f46dca6e203b0bdbb12c4aa9d79a6a3895b/src/bidiMapper/domains/context/cdpTarget.ts#L55).

Reply:
The easiest way to achieve that is to add a parameter `preloadScripts` to [`CdpTarget.create`](https://github.com/GoogleChromeLabs/chromium-bidi/blob/43882f46dca6e203b0bdbb12c4aa9d79a6a3895b/src/bidiMapper/domains/context/cdpTarget.ts#L36)

Reply:
Do you really mean `create` of CdpTarget, not of browsingContextImpl?

Reply:
it depends. From BiDi perspective it sticks to BrowserContext. And we tend to base structure on the BiDi. In other words, it can be both in Target and in BrowsingcontextProcessor. Up to you.

Reply:
I'm thinking if we should add e2e test which can expose such a race condition. E.g. add a preloadScript, and create a new tab with console.log(), and verify the event order.

Reply:
Will carry-over to a checkbox in #293